llvm/tools/repo-fragments/RepoFragments.cpp

```diff
@@ -0,0 +1,124 @@
+//===- repo-fragments - Dump the names and digests using a ticket file. --===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/MC/MCRepoTicketFile.h"
+#include "llvm/Support/CommandLine.h"
+
+#include "pstore/dump/mcrepo_value.hpp"
+
+using namespace llvm;
+
+namespace {
+
+static cl::OptionCategory ToolOptions("Tool Options");
+
+static cl::opt<std::string> RepoPath("repo", cl::Optional,
+                                     cl::desc("Path of the program repository"),
+                                     cl::init("./clang.db"),
+                                     cl::cat(ToolOptions));
+static cl::opt<std::string> TicketPath(cl::Positional,
+                                       cl::desc("<Path of the ticket file>"),
+                                       cl::Required, cl::cat(ToolOptions));
+static cl::list<std::string>
+    Names("names", cl::ZeroOrMore, cl::CommaSeparated,
```

Comment:
I think that `cl::CommaSeparated` specifies that the names will be comma-separated by the user rather than by repeating the ‑names switch. This is different to the python original. What were the reasons for this choice?

Reply:
>
>What were the reasons for this choice?
>

As you mentioned, I need to repeat  `-names` switch to give more fragment/compilation-member names.
`repo-fragments repo.o -names=func -names=var`.

It is more readable for me use comma in a single `-names`.
`repo-fragments repo.o -names=func,var`.

However, both formats are fine for me. Let me know if you prefer to remove `cl::CommaSeparated` .

Reply:
> Let me know if you prefer to remove cl::CommaSeparated .

What happens if CommaSeparated remains and the ‑‑names switch is used twice?

Reply:
> 
> What happens if CommaSeparated remains and the ‑‑names switch is used twice?

Very good question :). I tried the following two commands. 


```
$ repo-fragments repo.o -names=Var1 -names=Var2
$ repo-fragments repo.o -names=Var1,Var2
```
Both commands works and give the same output.
```
Var1: 5aaaba418d5cb1880ea77e7fea1c13ab
Var2: fcce730255dd6f37764eac7a6d3a4a59
```

Reply:
Great. I’m happy to leave it as is.